src/snowflake/cli/_plugins/nativeapp/entities/application.py

```diff
@@ -427,6 +441,71 @@ def get_objects_owned_by_application(self) -> List[ApplicationOwnedObject]:
             ).fetchall()
             return [{"name": row[1], "type": row[2]} for row in results]
 
+    def _should_authorize_event_sharing(
+        self,
+        install_method: SameAccountInstallMethod,
+        connection: SnowflakeConnection,
+        deploy_root: str,
+    ) -> Optional[bool]:
+        """
+        Logic to determine whether event sharing should be authorized or not.
+        If the return value is None, it means that authorize_event_sharing should not be updated.
+        If the return value is True/False, it means that authorize_event_sharing should be set to True/False respectively.
+        """
+
+        model = self._entity_model
+        workspace_ctx = self._workspace_ctx
+        console = workspace_ctx.console
+        project_root = workspace_ctx.project_root
+        is_dev_mode = install_method.is_dev_mode
+
+        mandatory_events_found = (
+            len(find_mandatory_events_in_manifest_file(project_root / deploy_root)) > 0
+        )
+        event_sharing_enabled = (
+            get_ui_parameter(connection, UIParameter.EVENT_SHARING_V2, "true").lower()
+            == "true"
+        )
+        event_sharing_enforced = (
+            get_ui_parameter(
+                connection, UIParameter.ENFORCE_MANDATORY_FILTERS, "true"
+            ).lower()
+            == "true"
+        )
+        authorize_event_sharing = (
+            model.telemetry and model.telemetry.authorize_event_sharing
+        )
+
+        if not event_sharing_enabled:
+            # We cannot set AUTHORIZE_TELEMETRY_EVENT_SHARING to True or False if event sharing is not enabled,
+            # so we will ignore the field in both cases, but warn only if it is set to True
+            if authorize_event_sharing:
+                console.warning(
+                    "WARNING: Same-account event sharing is not enabled in your account, therefore, application telemetry field will be ignored."
+                )
+            return None
+
+        if event_sharing_enabled and not event_sharing_enforced:
+            # if event sharing is enabled but not enforced yet, warn about future enforcement
+            if mandatory_events_found and not authorize_event_sharing:
+                console.warning(
+                    "WARNING: Mandatory events are present in the manifest file, but event sharing is not authorized in the application telemetry field. This will soon be required to set in order to deploy applications."
```

Comment:
```suggestion
                    "WARNING: Mandatory events are present in the manifest file, but event sharing is not authorized in the application telemetry field. This will soon be required to set in order to deploy this application."
```